lib/her/model/http.rb

```diff
@@ -97,7 +97,6 @@ def #{method}_resource(path, params={})
 
             def custom_#{method}(*paths)
               metaclass = (class << self; self; end)
-              opts = paths.last.is_a?(Hash) ? paths.pop : Hash.new
```

Comment:
Nice find 👍 Only risk may be if someone passed in options expecting them to be used, and then [L103](https://github.com/remiprev/her/blob/master/lib/her/model/http.rb#L103) failed. Another option could be 
```ruby
paths.pop if paths.last.is_a?(Hash)
```
That said, `custom_` options [doesn't](https://github.com/remiprev/her/blob/master/lib/her/model/http.rb#L14) [appear](https://github.com/remiprev/her#custom-requests) to be documented so perhaps it is ok to remove this line altogether. @edtjones any thoughts?

Reply:
I would prefer something non-breaking 🙂 - I have a vague recollection of using this method somewhere.

Reply:
(But we should document it if we are keeping it, and add some tests 🙂)

Reply:
Thanks for your responses.
Since these options aren't really used further in the method, should we also put some deprecation warning along with the line mentioned above? 
Something like:
```ruby
warn "[DEPRECATION] options for custom request methods are deprecated and will be removed in the next release."
paths.pop if paths.last.is_a?(Hash)
```

Reply:
That'd be awesome 😎 Can we issue the warning only if opts are passed in? And tweak the message to something like?

 ```
[DEPRECATION] options for custom request methods are deprecated and will be removed on or after January 2020.
```